traffic_portal/app/src/modules/private/cacheStats/CacheStatsController.js

```diff
@@ -17,60 +17,218 @@
+	 *
+	 * @returns {HTMLElement | Text} A rendered cell element.
+	 */
+	getGui() {
+		return this.eGui;
+	}
+}
 
-	$scope.ssh = serverUtils.ssh;
+/**
+ *
+ * @param {CacheStat[]} cacheStats
+ * @param {{user: {username: string}}} userModel
+ */
+var CacheStatsController = function(cacheStats, $scope, userModel) {
 
-	$scope.bandwidth = function(kbps, unit) {
-		return numberUtils.addCommas(numberUtils.convertTo(kbps, unit));
-	};
+	class CacheStatsSSHCellRenderer {
+		/** @type HTMLAnchorElement | Text */
+		eGui;
 
-	$scope.connections = function(amt) {
-		return numberUtils.addCommas(amt);
-	};
+		/**
+		 * Called by AG-Grid as a pseudo constructor, used when a renderer is
+		 * instantiated.
+		 *
+		 * @param {{
+		 * 	api: any;
+		 * 	colDef: any;
+		 * 	column: any;
+		 * 	columnApi: any;
+		 * 	context: any;
+		 * 	data: CacheStat
+		 * 	eGridCell: HTMLElement;
+		 * 	eParentOfValue: HTMLElement;
+		 * 	formatValue: function;
+		 * 	fullWidth: boolean;
+		 * 	getValue: function;
+		 * 	node: any;
+		 * 	pinned: string | null;
+		 * 	refreshCell: function;
+		 * 	registerRowDragger: function;
+		 * 	rowIndex: number;
+		 * 	setValue: function;
+		 * 	value: string;
+		 * 	valueFormatted: null;
+		 * 	$scope: null;
+		 * }} params
+		 */
+		init(params) {
+			if (params.data.ip === null) {
+				this.eGui = document.createTextNode(params.value);
+				return;
+			}
+			this.eGui = document.createElement("a");
+			this.eGui.href = `ssh://${userModel.user.username}@${params.data.ip}`;
+			this.eGui.setAttribute("class", "link");
+			this.eGui.textContent = params.value;
+		}
 
-	$scope.refresh = function() {
-		$state.reload(); // reloads all the resolves for the view
-	};
+		/**
+		 * Gets a rendered cell. Parameters are available, but not currently used.
+		 *
+		 * @returns {HTMLElement | Text} A rendered cell element.
+		 */
+		getGui() {
+			return this.eGui;
+		}
+	}
 
-	$scope.$on("$destroy", function() {
-		killInterval();
-	});
+	/** The columns of the ag-grid table */
+	$scope.columns = [
+		{
+			headerName: "Cache Group",
+			field: "cachegroup",
+			hide: false
+		},
+		{
+			headerName: "Connections",
+			field: "connections",
+			hide: false,
+			filter: "agNumberColumnFilter"
+		},
+		{
+			cellRenderer: CacheStatsHealthyCellRenderer,
+			headerName: "Healthy",
+			field: "healthy",
+			hide: false
+		},
+		{
+			cellRenderer: CacheStatsSSHCellRenderer,
+			headerName: "Host",
+			field: "hostname",
+			hide: false
+		},
+		{
+			cellRenderer: CacheStatsSSHCellRenderer,
+			headerName: "IP",
+			field: "ip",
+			hide: true,
+		},
+		{
+			headerName: "Data out rate",
+			field: "kbps",
+			filter: "agNumberColumnFilter",
+			hide: false,
+			/**
+			 * Formats the kbps metric.
+			 * @param {{value: number}} value Some AG Grid params containing the raw value to be formatted.
+			 * @returns {string} The formatted value.
+			 */
+			valueFormatter: ({value}) => {
+				console.log("formatting a kbps:", value);
```

Comment:
I don't think this is needed, but if you want to keep it it should probably be `console.debug`